File: stack/TwoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap map = new HashMap<Integer, Integer>();
        int[] res = new int[2];
        //iterate the array (nums)
        for (int i = 0; i < nums.length; i++ ) {
            int value = target - nums[i];
            if (map.containsKey(value)) {
                res[0] = (int)map.get(value);
                res[1] = i;
                return res;
            }
                
            map.put(nums[i], i);
        }
        res[0] = -1;
        res[1] = -1;
        return res;
    }
}

File: sliding_window/TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        appearances = {}
        for i in range(0, len(nums)):
            if nums[i] in appearances:
                continue
            appearances.update({nums[i], i})
            if (target - nums[i]) in appearances:
                return (i, appearances.get(target - nums[i]))
        
            
        